### ISSUE TITLE
There is a problem with the result of unpacking and packing boot.img.

### ISSUE THREAD

**Issue by lethargy123** [chenxiaolong/avbroot#144]
After unpacking a 64MB boot image, the generated file size is only 38.7MB, while the file generated by MagiskBoot is 39.9MB. Comparing the generated files, I noticed that avbroot did not generate the ramdisk.cpio file. I tried packing with avbroot, but the resulting file was also only 38.7MB. I used a binary comparison tool to compare the packed file and the original boot.img, and found that the beginning of the packed file is identical to the original boot.img, but after the 38.7MB mark, the data is missing. I believe this is either a bug or an issue with the implementation of this part of the packing process.
<img width="1065" alt="image" src="https://github.com/chenxiaolong/avbroot/assets/105811408/e67351a2-c0c9-4e46-ac5c-35898b07b5b0">

**Comment by chenxiaolong** [comment]
Based on your screenshot, this seems to be expected behavior, unless I'm misunderstanding what you're describing.

The `AVB0` bytes mark the beginning of the vbmeta header (and `AVBF` towards the end of the file marks the start of the vbmeta footer). The vbmeta header and footer are not preserved by `avbtool boot pack`. magiskboot takes a different approach and copies the original vbmeta header verbatim (which may leave it in an invalid state because checksums are not updated).

Because avbroot doesn't add the vbmeta information, it also does not pad the file size to 64MiB, which is only required when the vbmeta header + vbmeta footer are present.

(Also, it's worth noting that all of the actual boot image data, like the kernel and ramdisk, are always before `AVB0`.)

**Comment by chenxiaolong** [comment]
Support for unpacking and repacking AVB has been added in #152 and will be included in avbroot 2.1.0.

You'll be able to do this:

```bash
avbroot avb unpack -i boot.img
avbroot boot unpack -i raw.img

# Edit ramdisk or whatever...

avbroot boot pack -o raw.img
avbroot avb pack -o boot.img --key avb.key
```
